core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -536,11 +558,7 @@ sealed abstract class IO[+E, +A] extends Serializable { self =>
   final def timeout[B](z: B)(f: A => B)(duration: Duration): IO[E, B] =
     self
       .map(f)
-      .attempt
-      .raceWith[E, Either[E, B], B, B](IO.now[B](z).delay(duration))(
-        (either: Either[E, B], right: Fiber[E, B]) => right.interrupt *> either.fold(IO.fail, IO.now),
-        (b: B, left: Fiber[E, Either[E, B]]) => left.interrupt *> IO.now(b)
-      )
+      .raceWith(IO.now[B](z).delay(duration))(_.join <* _.interrupt, _.join <* _.interrupt)
```

Comment:
This breaks `timeout of failure` test... Is there a way to make `upTo` fail the test instead of skipping in specs2?

Reply:
Oh, I did not see it. Of course it fails 🤦‍♂️

Reply:
Hmm, wait, that's normal. If the first fiber fails, we'll wait for the second one. That's part of the semantics of `raceWith`.

Reply:
Fixed, thanks 👍